packages/paste-core/components/button/src/types.ts

```diff
@@ -4,9 +4,12 @@ export type ButtonVariants = 'primary' | 'secondary' | 'destructive' | 'destruct
 export type ButtonStates = 'disabled' | 'loading' | 'default';
 export type ButtonTabIndexes = 0 | -1;
 
-export interface ButtonProps {
+export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
```

Comment:
Because we’re now extending the button attributes into our interface, some of these declared props can probably just be removed as they would be present on the native. 

Unless we override anything with that prop, like make it required, we can just safely rely on the button element interface. We’re trying to make the button feel like a native HTML button after all. 

I’d go as far as to remove any opinions we have for things like `aria-haspopup` and just rely on the button element interface

Reply:
Cleaned up the props based on what we get from the native HTML button attributes. Let me know if you see anything I missed.